testnet/stacks-node/src/nakamoto_node/relayer.rs

```diff
@@ -916,7 +906,7 @@ impl RelayerThread {
             "last_winner_snapshot.miner_pk_hash" => ?last_winner_snapshot.miner_pk_hash,
             "canonical_stacks_tip_id" => %canonical_stacks_tip,
             "canonical_stacks_tip_ch" => %canonical_stacks_tip_ch,
-            "block_election_ch" => %block_election_snapshot.consensus_hash,
+            "block_election_ch" => %last_winner_snapshot.consensus_hash,
```

Comment:
Why was the previous code incorrect?  If the node doesn't have some intermediate blocks, then `last_winner_snapshot` may not correspond to a tenure this node can build on.

Reply:
Right -- in the scenarios in which the prior code is incorrect, the signer set probably wouldn't let this miner build via a tenure extend _anyways_, so this patch is pretty low impact. The scenario in which this was uncovered was a failed test run of the `partial_tenure_forking` test.

In the test failure, a sortition C was won by the null miner (triggering the empty sortition logic in `nakamoto_node::relayer`). The prior sortition (Sortition B) _was_ won by the miner, but the latest chain tip was in a sortition prior to that one (i.e., Sortition A). So, the logic here told the miner thread to wake up and try to issue an extension with consensus hash corresponding to Sortition A, which was no longer in the .miners stackerdb. Instead, the miner should be attempting to continue Sortition B (but even that wouldn't work, because no blocks were accepted in Sortition B yet, so the tenure extend would get rejected). Sortition B didn't have any blocks produced in it because the test explicitly forced the miner into initiating a fork that the signer set rejects.

Reply:
Okay, as long as you're sure that the signers won't let the miner build via a tenure-extend on A (but it seems the .miners contract forbids it anyway).

Reply:
Can you add a comment to the effect of the above?

Reply:
Sure can do.

Reply:
Actually I don't fully undertasnd what the comment would be because I don't thnk I understand the fix properly...

Reply:
I think it would be correct to say:

```
// The node will try to build on the sortition with the last winner, even if it does not yet have all of
// the ancestor Stacks blocks necessary to do so.  This is okay, because (1) the node eventually
// will catch up and (2) the signers won't let the miner orphan blocks they have already seen (and
// signers see all the blocks).  So, the miner should try and mine off of the chain tip the signers 
// will accept, if it mines at all.
```